src/lib/comms/sol-http-common.c

```diff
@@ -77,19 +77,17 @@ sol_http_param_free(struct sol_http_param *params)
 }
 
 SOL_API int
-sol_http_escape_string(char **escaped, const char *value)
+sol_http_escape_string(char **escaped, const struct sol_str_slice value)
 {
     int r;
-    size_t value_len;
+    size_t i;
     struct sol_buffer buf = SOL_BUFFER_INIT_EMPTY;
 
-    SOL_NULL_CHECK(value, -EINVAL);
     SOL_NULL_CHECK(escaped, -EINVAL);
 
-    value_len = strlen(value);
 
-    while (value_len--) {
-        unsigned char c = *value;
+    for (i = 0; i < value.len; i++) {
+        unsigned char c = value.data[i];
         switch (c) {
         case '0': case '1': case '2': case '3': case '4':
```

Comment:
ugh, not related to your patch, but this switch is ugly. In a future patch we could simplify it with `isalnum()` (0-9, a-z, A-Z) followed by the extra digits we allow, or we do `if ('0' <= c && c <= '9') ... ` explicitly... or we use the same trick as @lpereira did with JSON scanner that uses `strchr("0123456789abc..._~.-", c)`that will search if the string contains the current char. I like `isalnum()` better.